File: java/src/test/java/com/shipzy/sdk/EpodClientTest.java
package com.shipzy.sdk;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class EpodClientTest {

    @Test
    void constructor_withDefaultConfig_works() {
        EpodClient.ShipzyConfig config = new EpodClient.ShipzyConfig();
        EpodClient client = new EpodClient(config);
        assertNotNull(client);
    }

    @Test
    void constructor_withCustomConfig_works() {
        EpodClient.ShipzyConfig config = new EpodClient.ShipzyConfig();
        config.setBaseUrl("http://localhost:1417");
        config.setToken("test-token");
        config.setTimeoutSeconds(60);
        EpodClient client = new EpodClient(config);
        assertNotNull(client);
    }

    @Test
    void setToken_updatesToken() {
        EpodClient.ShipzyConfig config = new EpodClient.ShipzyConfig();
        EpodClient client = new EpodClient(config);
        assertDoesNotThrow(() -> client.setToken("new-token"));
    }

    @Test
    void config_defaultValues_areCorrect() {
        EpodClient.ShipzyConfig config = new EpodClient.ShipzyConfig();
        assertEquals("https://api.shipzy.me", config.getBaseUrl());
        assertNull(config.getToken());
        assertEquals(30, config.getTimeoutSeconds());
    }

    @Test
    void config_canOverrideValues() {
        EpodClient.ShipzyConfig config = new EpodClient.ShipzyConfig();
        config.setBaseUrl("http://localhost:1417");
        config.setToken("my-token");
        config.setTimeoutSeconds(60);
        assertEquals("http://localhost:1417", config.getBaseUrl());
        assertEquals("my-token", config.getToken());
        assertEquals(60, config.getTimeoutSeconds());
    }

    @Test
    void shipzyException_hasStatusCode() {
        ShipzyException ex = new ShipzyException("test error", 400);
        assertEquals(400, ex.statusCode);
        assertEquals("test error", ex.getMessage());
    }

    @Test
    void shipzyAuthException_is401() {
        ShipzyAuthException ex = new ShipzyAuthException("unauthorized");
        assertEquals(401, ex.statusCode);
    }
}

File: kotlin/src/test/kotlin/com/shipzy/sdk/EpodClientTest.kt
package com.shipzy.sdk

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFailsWith
import kotlin.test.assertNotNull
import kotlin.test.assertNull

class EpodClientTest {

    @Test
    fun constructor_withDefaultConfig_works() {
        val config = ShipzyConfig()
        val client = EpodClient(config)
        assertNotNull(client)
    }

    @Test
    fun constructor_withCustomConfig_works() {
        val config = ShipzyConfig(
            baseUrl = "http://localhost:1417",
            token = "test-token",
            timeoutSeconds = 60
        )
        val client = EpodClient(config)
        assertNotNull(client)
    }

    @Test
    fun setToken_updatesToken() {
        val config = ShipzyConfig()
        val client = EpodClient(config)
        client.setToken("new-token")
        // No exception
    }

    @Test
    fun config_defaultValues_areCorrect() {
        val config = ShipzyConfig()
        assertEquals("https://api.shipzy.me", config.baseUrl)
        assertNull(config.token)
        assertEquals(30, config.timeoutSeconds)
    }

    @Test
    fun config_canOverrideValues() {
        val config = ShipzyConfig(
            baseUrl = "http://localhost:1417",
            token = "my-token",
            timeoutSeconds = 60
        )
        assertEquals("http://localhost:1417", config.baseUrl)
        assertEquals("my-token", config.token)
        assertEquals(60, config.timeoutSeconds)
    }

    @Test
    fun shipzyException_hasStatusCode() {
        val ex = ShipzyException("test error", 400)
        assertEquals(400, ex.statusCode)
        assertEquals("test error", ex.message)
    }

    @Test
    fun shipzyAuthException_is401() {
        val ex = ShipzyAuthException("unauthorized")
        assertEquals(401, ex.statusCode)
    }
}
